ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -194,12 +194,19 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			err = r.updateState(ctx, rayJobInstance, nil, rayJobInstance.Status.JobStatus, rayv1alpha1.JobDeploymentStatusWaitForDashboard, err)
 			return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 		}
+		// Check the dashboard readiness by checking the err return from rayDashboardClient.GetJobInfo.
+		// Note that rayDashboardClient.GetJobInfo returns no error in the case of http.StatusNotFound.
+		// This check is a workaround for https://github.com/ray-project/kuberay/issues/1381.
+		rayDashboardClient.InitClient(clientURL)
+		if _, err = rayDashboardClient.GetJobInfo(ctx, rayJobInstance.Status.JobId); err != nil {
```

Comment:
Out of curiosity why use getJobInfo(jobID)? The job hasn't been submitted yet, so I guess there won't be any jobInfo available yet.  Is this just an arbitrary HTTP request to make sure the server is working?

Reply:
> Is this just an arbitrary HTTP request to make sure the server is working?

Yes. Can we conclude that the dashboard is ready for the `ray job submit` command if `GetJobInfo` succeeds?

Reply:
Yeah, I think that's a fair conclusion

Reply:
Yes, it is just an arbitrary HTTP request to make sure the server is working.

So it is ok to go proceed with the job submission if we get 404 from the GetJobInfo here. That means the dashboard is working.

Reply:
> So it is ok to go proceed with the job submission if we get 404 from the GetJobInfo here. That means the dashboard is working.

Does this imply that a 404 response will be returned if the JobID does not exist?

Reply:
Maybe not. But a 404 response implies the dashboard server is up.